presto-hive/src/test/java/com/facebook/presto/hive/HiveTestUtils.java

```diff
@@ -65,29 +66,34 @@ private HiveTestUtils()
         return types.build();
     }
 
-    public static Slice arraySliceOf(Type elementType, Object... values)
+    public static Block arraySliceOf(Type elementType, Object... values)
     {
-        BlockBuilder blockBuilder = new VariableWidthBlockBuilder(new BlockBuilderStatus(), 1024);
+        BlockBuilder blockBuilder = elementType.createBlockBuilder(new BlockBuilderStatus(), 1024);
         for (Object value : values) {
             appendToBlockBuilder(elementType, value, blockBuilder);
         }
-        return buildStructuralSlice(blockBuilder);
+        return blockBuilder.build();
     }
 
-    public static Slice mapSliceOf(Type keyType, Type valueType, Object key, Object value)
+    public static Block mapSliceOf(Type keyType, Type valueType, Object key, Object value)
     {
-        BlockBuilder blockBuilder = new VariableWidthBlockBuilder(new BlockBuilderStatus(), 1024);
+        BlockBuilder blockBuilder = new InterleavedBlockBuilder(ImmutableList.of(keyType, valueType), new BlockBuilderStatus(), 1024);
         appendToBlockBuilder(keyType, key, blockBuilder);
         appendToBlockBuilder(valueType, value, blockBuilder);
-        return buildStructuralSlice(blockBuilder);
+        return blockBuilder.build();
     }
 
-    public static Slice rowSliceOf(List<Type> parameterTypes, Object... values)
+    public static Block rowSliceOf(List<Type> parameterTypes, Object... values)
     {
-        BlockBuilder blockBuilder = new VariableWidthBlockBuilder(new BlockBuilderStatus(), 1024);
+        ArrayBlockBuilder blockBuilder = new ArrayBlockBuilder(
```

Comment:
you can remove the `ArrayBlockBuilder` and just use an `InterleavedBlockBuilder` then return `.build()` instead of calling `getObject()`